### ISSUE TITLE
mb_regex_set_options never had eval option?

### ISSUE THREAD

**Issue by afilina** [php/doc-en#2103]
- PHP 7.1: Eval option deprecated in `mb_ereg_replace` and `mb_eregi_replace` ([RFC](https://wiki.php.net/rfc/deprecate_mb_ereg_replace_eval_option)).
- PHP 8.0: Eval option removed, but also throws a `ValueError` with `mb_regex_set_options` (not documented, AFAIK).

However, when trying to set `e` using `mb_regex_set_options` in versions where it was still supported, it seems to have no effect.

```php
// In PHP 5.0 through 7.4, this returns ixplnr, without e
var_dump(mb_regex_set_options('ixplner')); 
// this does not have the eval effect
var_dump(mb_ereg_replace('(h)', 'strtoupper("\\1")', 'hello', null)); 
```

Result in all versions: https://3v4l.org/pXd41G

Was eval ever a thing with `mb_regex_set_options`? Should I update the docs to reflect reality?

**Comment by damianwadley** [comment]
At least as of 7.4, the "e" option only worked when passed during a call to mb_ereg(i)_replace. It was not a persistent setting like the others (very sensibly so, I would say) and thus doesn't do anything if provided to mb_regex_set_options. That's also why you don't see it returned after the set.

The documentation overall isn't covering "e" very well - not just with mb_regex_set_options but the other affected functions too. In summary,
- "e" was deprecated in 7.4 and removed with 8.0
- Since 8.0, unrecognized options passed to any of the functions results in a ValueError
- "e" worked only with mb_ereg_replace and mb_eregi_replace, and did nothing when passed to other functions

**Comment by afilina** [comment]
Thank you for the reply. If I understand correctly, "e" is documented on the `mb_regex_set_options` page, but in reality, it's only an option that could be set via the options argument of the `mb_ereg(i)_replace` functions. Correct?

**Comment by damianwadley** [comment]
Correct, in that you could set "e" however it wouldn't have any effect. Because the same code was (and still is) being used to support the options when they're passed to mb_regex_set_options as well as to mb_ereg(i)_replace and a couple other functions too.
With the removal of "e" in 8.0, all of the functions now behave the same way and support the same options.
